tests/unit/recommenders/evaluation/test_python_evaluation.py

```diff
@@ -164,10 +150,10 @@ def test_python_rmse(rating_true, rating_pred):
         )
         == 0
     )
-    assert rmse(rating_true, rating_pred) == pytest.approx(7.254309, TOL)
+    assert rmse(rating_true, rating_pred) == target_metrics["rmse"]
```

Comment:
In other tests we hard code the results, see https://github.com/recommenders-team/recommenders/tree/main/tests#how-to-create-tests-for-the-recommenders-library with the pattern computation equals value.

I would argue it's more clear and explicit

Reply:
I agree with you when an expected value is clear and easy to understand. E.g.
```
# check if the metric is the best number (rmse == 0) if the prediction is perfect, i.e true == prediction.
assert rmse(rating_true, rating_true) == 0
```

I also agree that it is explicit to put the number directly when we also specify the input in the same way, like:
```
output = subtract_two(2.2345, 1.0)  
assert output == 1.2345  # this is explicit

# versus

output = subtract_two(input_fixture_data_a, input_fixture_data_b)
assert output == 1.2345  # our input is already in a blackbox for reusability
```

So, I still prefer using an expected value fixture, the same reason we use the fixture for the input data:
1. By using a single fixture value, we can reuse them across many tests and easy to maintain the test cases with the expected values.
2. We can assure both python and spark implementation produces the same metrics for the same inputs (in addition to the test we directly check both metrics' equalities, `test_spark_python_match`)

What do you think?

Reply:
> I still prefer using an expected value fixture, the same reason we use the fixture for the input data

There are several things to consider here:
- Fixtures are not designed for the results of computations it is designed for data. Because you want to control how the data is loaded (test level, module, or session). For example, maybe we want to load movielens at a module level and not every test because it adds 1 or 2 seconds for every test. A result of a computation doesn't need that, because it is just one number.
- The pattern should be computation == value. Because a common failure is to do a computation1 = computation2, people don't realize that computation2 can also fail, so then they think they are testing computation1, but the error can come from computation1 or computation2. I think it is important to make clear this to less experienced developers (after all, we are trying to teach good coding practices).
- I agree with you that it is convenient to have common results for python and spark, but then it shouldn´t be a fixture, but constants. We can add them as constants or as a dict, or we can directly add the number to keep the consistency with the rest of the repo. 

Thoughts?

Reply:
@miguelgfierro per your suggestion, I changed the way the tests check values. Please review again and see if this PR is good to merge!